collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/core/json/TestRunJsonParser.java

```diff
@@ -0,0 +1,106 @@
+package com.capitalone.dashboard.core.json;
+
+import com.atlassian.jira.rest.client.internal.json.JsonObjectParser;
+import com.atlassian.jira.rest.client.internal.json.JsonParseUtil;
+import com.capitalone.dashboard.api.domain.Comment;
+import com.capitalone.dashboard.api.domain.TestRun;
+import com.capitalone.dashboard.api.domain.TestStep;
+import com.capitalone.dashboard.core.json.util.XrayJiraDateFormatter;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * This class will parse the JSON for a Test Run
+ */
+public class TestRunJsonParser implements JsonObjectParser<TestRun> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestRunJsonParser.class);
+
+    public final static String KEY_ID="id";
+    public final static String KEY_STATUS="status";
+    public final static String KEY_EXECBY="executedBy";
+    public final static String KEY_ASSIGNEE="assignee";
+    public final static String KEY_STARTEDON="startedOn";
+    public final static String KEY_FINISHEDON="finishedOn";
+    public final static String KEY_EXAMPLES="examples";
+    public final static String KEY_COMMENT="comment";
+    public final static String KEY_TESTSTEPS="steps";
+    // TODO: CUMCUMBER TESTS
+    // public final static String KEY_SCENARIO="scenario";
+    // public final static String KEY_SCENARIO_OUTLINE="scenarioOutline";
+
+
+    public TestRunJsonParser(){
+    }
+
+    public TestRun parse(JSONObject jsonObject) throws JSONException {
+        jsonObject.put("self",""); // TODO: ADD URI.
+        URI selfUri = JsonParseUtil.getSelfUri(jsonObject);
+        String key =" THERE IS NO KEY FOR TEST RUN AT X-RAY DIRECT REST API"; // TODO: GET THE ISSUE KEY
```

Comment:
@kumarvarun1252 key value is assigned as  THERE IS NO KEY FOR TEST RUN AT X-RAY DIRECT REST API and never re-assigned. is this value passed to create TestRun

Reply:
This is an optional field and it doesn't affect parsing TestRun. Added TODO